authentication.md

```diff
@@ -16,21 +16,21 @@ The framework integration will provide the application with some means of resolv
 
 * Reading the request headers for authentication information
 * Resolving the access token or API key presented to a Stormpath account
-* Return this information to the application, along with the method of authentication.
+* Return this information to the application, along with information about how the request was authenticated.
 
 This information will allow the application to make authorization decisions for the request. 
 
 **Unauthenticated Requests**
 
-In addition, the framework integration MAY provide the application with some means of rendering errors for unauthenticated requests. This may be combined with the Account Resolvers for extra convenience. 
+As a convenience, the framework integration MAY provide the application with a means to, in one step, check for authentication information and render an error if the request is unauthenticated. 
```

Comment:
This is optional, so it's more for implementations that support this sort of model. 

There are really two parts to this PR: 

* **Account Resolver**: 
 * Is the request authenticated? If so, return who and how
* **Optional**: Unauthenticated request behavior: 
 * Is the request authenticated? If not, render error. If so, return who and how

If we split the unauthenticated behavior into multiple steps, then we have a helper that's just a renderer. And I thought that was explicitly what people did not want at the meeting?